### PR TITLE
Update server-base and server-slim-base to production-like images

### PR DESCRIPTION
**Pull request by devinrsmith**
This was a longer-than-expected effort to close the gap between production images, and integration testing images. The big updates: the base images are now based on ubuntu:22.04 (instead of an old debian), and the python version is now 3.10 (instead of an old 3.7). These images are now being produced with Dockerfiles from https://github.com/deephaven/deephaven-server-docker that are as similar to the production images as possible.

Necessary testing fixes already done in #3331

### CODE REVIEW

**Comment by devinrsmith** [comment]
Looks like some sphinx typing/documentation issues w/ 3.10.

**Comment by jmao-denver** [comment]
I was able to make the sphinx task pass locally with the following change.
```
diff --git a/sphinx/source/conf.py b/sphinx/source/conf.py
index 5f3ee1b4c9..e8116e9c9a 100644
--- a/sphinx/source/conf.py
+++ b/sphinx/source/conf.py
@@ -85,10 +85,26 @@ jvm_properties = {
             'deephaven.dataDir': os.path.realpath(workspace),
         }

+jvm_options = {
+    '-XX:InitialRAMPercentage=25.0',
+    '-XX:MinRAMPercentage=70.0',
+    '-XX:MaxRAMPercentage=80.0',
+
+    # Allow netty to (reflectively) access java.nio.Buffer fields
+    '--add-opens=java.base/java.nio=ALL-UNNAMED',
+
+    # Allow our hotspot-impl project to access internals
+    '--add-exports=java.management/sun.management=ALL-UNNAMED',
+
+    # Allow our clock-impl project to access internals
+    '--add-exports=java.base/jdk.internal.misc=ALL-UNNAMED',
+}
+
 from deephaven_internal import jvm
 jvm.init_jvm(
     jvm_classpath=glob(os.environ.get('DEEPHAVEN_CLASSPATH')),
     jvm_properties=jvm_properties,
+    jvm_options=jvm_options,
 )

 import jpy
@@ -96,6 +112,7 @@ py_scope_jpy = jpy.get_type("io.deephaven.engine.util.PythonScopeJpyImpl").ofMai
 py_dh_session = jpy.get_type("io.deephaven.integrations.python.PythonDeephavenSession")(py_scope_jpy)
 py_dh_session.getExecutionContext().open()

+
 import deephaven
 docs_title = "Deephaven python modules."
 package_roots = [jpy, deephaven]
```

**Comment by devinrsmith** [comment]
I also needed to update the location of grpc_health_probe to match the production location

**Review by jmao-denver — APPROVED** [review]
LGTM and partially tested locally

File: py/jpy-integration/src/pythonToJava/python/test/test_jpy.py [hunk 1]
```diff
@@ -118,13 +118,16 @@ def test_numpy_array(self):
...
     self.assertEqual(get_refcount(jpy_array_id), jpy_array_refcount)
 
-    mv = memoryview(b'123412341234')
-    mv_id = id(mv)
-    mv_refcount = get_refcount(mv_id)
-    np_array = numpy.frombuffer(mv, numpy.int32)
-    self.assertEqual(get_refcount(mv_id), mv_refcount + 1)
-    np_array = None
-    self.assertEqual(get_refcount(mv_id), mv_refcount)
+    # The refcounting seems to work different in the memoryview case as we update from 3.7 to 3.10; either cpython or
```
Comment (rcaudy) [inline py/jpy-integration/src/pythonToJava/python/test/test_jpy.py:121]: Delete me?

File: sphinx/source/conf.py [hunk 2]
```diff
@@ -85,17 +85,34 @@
             'deephaven.dataDir': os.path.realpath(workspace),
         }
 
+jvm_options = {
+    '-XX:InitialRAMPercentage=25.0',
+    '-XX:MinRAMPercentage=70.0',
+    '-XX:MaxRAMPercentage=80.0',
+
+    # Allow netty to (reflectively) access java.nio.Buffer fields
```
Comment (rcaudy) [inline sphinx/source/conf.py:93]: Is this valid? Only netty? Does sphinx still use netty?

Comment (devinrsmith) [reply sphinx/source/conf.py]: I think the netty part is too specific. It looks like it's also required by arrow, and potentially other low-level components. I've removed the explicit reference to "netty".

**Review by jmao-denver — APPROVED** [review]

**Review by rcaudy — APPROVED** [review]

**Pull request merged**